src/simplewallet/simplewallet.cpp

```diff
@@ -873,65 +873,86 @@ bool simple_wallet::set_log(const std::vector<std::string> &args)
-    }
-  }
-  while (!valid_path);
-
+  std::string confirm_creation;
+  bool wallet_name_valid = false;
   bool keys_file_exists;
   bool wallet_file_exists;
-  tools::wallet2::wallet_exists(wallet_path, keys_file_exists, wallet_file_exists);
-  LOG_PRINT_L3("wallet_path: " << wallet_path << "");
-  LOG_PRINT_L3("keys_file_exists: " << std::boolalpha << keys_file_exists << std::noboolalpha
-    << "  wallet_file_exists: " << std::boolalpha << wallet_file_exists << std::noboolalpha);
-
-  LOG_PRINT_L1("Loading wallet...");
 
-  // add logic to error out if new wallet requested but named wallet file exists
-  if (keys_file_exists || wallet_file_exists)
-  {
-    if (!m_generate_new.empty() || m_restoring)
+  do{
+      LOG_PRINT_L3("Request user input");
+      wallet_path = command_line::input_line(
+      tr("Specify wallet file name (e.g., MyWallet). If the wallet doesn't exist, it will be created.\n"
+         "Wallet file name (or Ctrl-C to quit): ")
+       );
+    if(std::cin.eof())
     {
-      fail_msg_writer() << tr("attempting to generate or restore wallet, but specified file(s) exist.  Exiting to not risk overwriting.");
+      LOG_ERROR("Unexpected std::cin.eof() - Exited simple_wallet::ask_wallet_create_if_needed()");
       return false;
     }
-  }
-
-  bool r;
-  if(keys_file_exists)
-  {
-    m_wallet_file=wallet_path;
-    r = true;
-  }else
-  {
-    if(!wallet_file_exists)
+    if(wallet_path.empty()) //Check for empty name
     {
-      std::cout << tr("The wallet doesn't exist, generating new one") << std::endl;
-      m_generate_new = wallet_path;
-      r = true;
-    }else
+      fail_msg_writer() << tr("Empty wallet names are invalid. Please try again or use Ctrl-C to quit.");
+      LOG_ERROR("Empty wallet names are invalid.");
+      wallet_name_valid = false;
+    }
+    else
     {
-      fail_msg_writer() << tr("keys file not found: failed to open wallet: ") << "\"" << wallet_path << "\".";
-      r = false;
+      tools::wallet2::wallet_exists(wallet_path, keys_file_exists, wallet_file_exists);
+      LOG_PRINT_L3("wallet_path: " << wallet_path << "");
+      LOG_PRINT_L3("keys_file_exists: " << std::boolalpha << keys_file_exists << std::noboolalpha
+        << "  wallet_file_exists: " << std::boolalpha << wallet_file_exists << std::noboolalpha);
+
+      if((keys_file_exists || wallet_file_exists) && (!m_generate_new.empty() || m_restoring))
+      {
+        fail_msg_writer() << tr("Attempting to generate or restore wallet, but specified file(s) exist.  Exiting to not risk overwriting.");
+        LOG_ERROR("Wallet/key file exists when asked to restore or generate new. Exiting.");
+        return false;
+      }
+      if(wallet_file_exists && keys_file_exists) //Yes wallet, yes keys
+      {
+        std::cout << tr("Wallet and key files found, loading...") << std::endl;
+        LOG_PRINT_L1("Wallet and key files found, loading...");
```

Comment:
Same as other calls in simplewallet, eg:

      fail_msg_writer() << tr("keys file not found: failed to open wallet: ") << "\"" << wallet_path << "\".";

The object (either fail_msg_writer or success_msg_writer, might be a third one IIRC), with parameters via the << operator.